File: c codes/Stack.c
#include<stdio.h>
#define MAX 5
int Stack[MAX];
int top=-1;

void push()
{
    int x;
    while(top<MAX)
    {
        if(top==MAX-1)
        {
            printf("MAX limit reached (Overflow)\n");
            return;
        }
        else
        {
            printf("Enter data\n");
            scanf("%d",&x);
            top++;
            Stack[top]=x;
        }
        
    }
}
void display()
{
    int i;
    printf("Your elements are\n");
    for(i=top;i>-1;i--)
    {
        printf("%d\n",Stack[i]);
    }
}
void peek()
{
    if(top==-1)
    {
        printf("Underflow\n");
    }
    else
    {
        printf("Topmost element is %d",Stack[top]);  
    } 
}
void pop()
{
    int item;
    if(top==-1)
    {
        printf("Underflow");
    }
    else
    {
        item=Stack[top];
        top--;
        printf("poped item is %d",item);
    }
    
}
void main()
{
    int a;

    while(a!=5)
    {
        printf("\nType 1 to push\nType 2 to Display\nType 3 to pop\nType 4 to peek\nType 5 to exit\n");
        scanf("%d",&a);
        switch (a)
        {
        case 1:
        push();
        break;
        
        case 2:
        display();
        break;
        
        case 3:
        pop();
        break;

        case 4:
        peek();
        break;

        case 5:
        printf("Exited\n");
        break;

        default:
        printf("Wrong Choice!!");
        }
    }
}
//Ashutosh Pandey

File: python-codes/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):

        return self.items.append(item)
        

    def pop(self):
        if self.is_empty():
            return print("Stack is Empty")
        return self.items.pop()
        

    def is_empty(self):
        return self.items == []

    def stack_length(self):
        return len(self.items)

    def peek(self):
        if self.is_empty():
            return print("Stack is Empty")
        return self.items[-1]
        


def main():
    stack = Stack()
    print("Type Your Name : ")
    name = input()
    print("Welcome " + name)
    print("Choose your option")
    while True:
        print(" 1. Push \n 2. Pop \n 3. Length \n 4. Top-Item \n 5. Show Stack")
        print("________________________\n________________________")
        choice = input()
        if choice == "1":
            print("Enter Element (Any Data Type)")
            x = input()
            stack.push(x)
            print("*********************\n*********************")
        elif choice == "2":
            stack.pop()
        elif choice == "3":
            print('Length of the Stack is in Below')
            print(stack.stack_length())
            print("*********************\n*********************")
        elif choice == "4":
            print("Next Pop Item is in below")
            print(stack.peek())
            print("*********************\n*********************")
        elif choice == "5":
            print(stack.items)
            print("*********************\n*********************")


main()
